source/client/process_impl.cc

```diff
@@ -179,7 +181,10 @@ void ProcessImpl::createWorkers(const uint32_t concurrency) {
   // TODO(oschaaf): Arguably, this ought to be the job of a rate limiter with awareness of the
   // global status quo, which we do not have right now. This has been noted in the
   // track-for-future issue.
-  const auto first_worker_start = time_system_.systemTime() + kMinimalWorkerDelay;
+  const Envoy::MonotonicTime monotonic_now = time_system_.monotonicTime();
```

Comment:
Three comments here:
1.  This has gotten complex enough (especially with the long comments above to explain what's going on), that I think it might warrant its own privately scoped function.  CalculateWorkerStartTime_ or something?